### ISSUE TITLE
Where to get ids for location, prices to order storages?

### ISSUE THREAD

**Issue by ematpoc** [softlayer/softlayer-go#178]
Hi, 
I want to create an app to order storages. Looking at the example here https://sldn.softlayer.com/go/order_endurance_storage.go/:
```
//To get the list of valid prices for the package use the SoftLayer_Product_Package:getItemPrices method
prices := []datatypes.Product_Item_Price {
		{ Id: sl.Int(45098) },  // Block Storage
		{ Id: sl.Int(45058) },  // Endurance Storage
		{ Id: sl.Int(45148) },  // 40 GB Storage Space
		{ Id: sl.Int(45068) },  // 0.25 IOPS per GB
	}
```
Where  those ids are taken from? How to get Ids for File Storage, Performance Storage, different Storage Spaces, Iops and locations, etc?
I was trying to call getItemPrices hoping that it gives me list of prices but it looks like I do it wrong:
```
svc := services.GetProductPackageService(sess)
pricing, err := svc.GetItemPrices()

but I got this:
SoftLayer_Exception: Object does not exist to execute method on. (SoftLayer_Product_Package::getItemPrices) (HTTP 200)

```

Could you help out? 
BR

**Comment by BrianSantivanez** [comment]
Hi @ematpoc, to get the item prices ID's is correct you are using SoftLayer_Product_Package::getItemPrices but you also need the package ID, in https://sldn.softlayer.com/go/order_endurance_storage.go/ you have this declaration 
```
// Declare the location, packageId and quantity for the storage you wish to order
	quantity := 1
	location := "MEXICO"
	packageId := 240
	osKeyName := "LINUX"
```
where it say the package ID to an Endurance Storage is 240 and you only have to add the ID in the request
```
svc := services.GetProductPackageService(sess)
pricing, err := svc.Id(240).GetItemPrices()
```

If you want to get the Item Prices to another package (File Storage, Performance Storage) you need the package ID of the new product package and to do it you can use SoftLayer_Product_Package::getAllObjects
```
service := services.GetProductPackageService(sess)
packages, err := service.GetAllObjects()
```

About the locations you can use SoftLayer_Product_Package::getRegions to know what Package is available in what locations
```
service := services.GetProductPackageService(sess)
locations, err := service.Id(240).GetRegions()
```

**Comment by ematpoc** [comment]
Thank you for your response. 
I find it extremely  difficult to navigate through all different kinds of options. 

Let's say I want to order storage with these parameters:
```
Storage as a Service
block storage
storage type: performance
datacenter: fra02
hourly_billing_flag: true
size: 40 GB
iops: 300
snapshot size: 40  GB
```


In Python library it's very simple, I call order_block_volume('performance', 'fra02', 40, 300, 40, true) and that's it.


I know that packageID for Storage As a Service is 759.
```
prices, err := svc.Id(759).GetItemPrices()
```

how to navigate through 'prices' to find  IDs for above parameters? 

BR

**Comment by allmightyspiff** [comment]
@ematpoc I got an email about a response you posted here, but I don't see the actual comment... but you had some good questions so I'll try to answer them.


> In Python library it's very simple, I call order_block_volume('performance', 'fra02', 40, 300, 40, true) and that's it.
While this library is sort of bare-bones, I don't mind adding helper functions if it would be used, let me know if that would help you out here.

In any case, the `ibmcloud sl` command works basically the same as the softlayer-python library but written in go. https://github.com/softlayer/softlayer-cli/blob/master/plugin/managers/storage.go#L528 is the equivalent function that might help you.

> how to navigate through 'prices' to find IDs for above parameters?
The following articles are my attempts at explaining how ordering works, hopefully they will help.
- https://sldn.softlayer.com/article/step_by_step_ordering/
- https://sldn.softlayer.com/article/understanding-ordering-basics/
- https://sldn.softlayer.com/article/catalog/
- https://sldn.softlayer.com/article/understanding-ordering/

Basically each Item in a Package has Prices, and its these Price Ids you need to send into the API. A single Item will have multiple Prices due to differing locations and restrictions. When placing an order, I always recommend sending in the default price id (its LocationGroupId is null).  

I hope that helps.
